### DISCUSSION TITLE
Help, how can I generate .cs file directly into the output Folder ?

### DISCUSSION THREAD

**Discussion by tuan3451** [mganss/XmlSchemaClassGenerator#536] [Q&A]
Hi,

First of all, thank you very much for such an amazing project. It has helped me a lot.

One problem I encountered is that when I select an output location, it generates a folder with the namespace name and then puts all the .cs files inside. I wonder if there is any option to prevent this, as I only want it to put the generated .cs files directly into my chosen output folder.

Thank you very much.

**Comment by mganss** [comment]
There are the following options that control output, though both of these are false by default so I'm surprised that you're getting a separate folder for the namespace.

```
      --sf, --separateFiles  generate a separate file for each class (default
                               is false)
      --nh, --namespaceHierarchy
                             generate a separate folder for namespace hierarchy.
                                Implies "separateFiles" if true (default is
                               false)
```

You can try passing `--sf- --nh-` to force both options to false though I doubt this will help. If it still doesn't work, can you provide a minimal example xsd?

**Reply by tuan3451** [reply]
Thank you for replying so quickly. It helped me realize why my file generated a structure is like that.

Currently, I created this function to set options:

```csharp
private static Generator GetGenerator(string nameSpace, string outputFolderPath)
{
    return new Generator
    {
        OutputFolder = outputFolderPath,
        GenerateNullables = true,
        SeparateClasses = true,
        CollectionType = Type.GetType("System.Collections.Generic.List`1"),
        CollectionImplementationType = Type.GetType("System.Collections.Generic.List`1"),
        CollectionSettersMode = CollectionSettersMode.PublicWithoutConstructorInitialization,
        GenerateInterfaces = false,
        GenerateComplexTypesForCollections = false,
        DisableComments = true,
        SeparateNamespaceHierarchy = false,
        NamespaceProvider = new NamespaceProvider { GenerateNamespace = key => nameSpace },
    };
}
```

As you can see, I set `SeparateClasses = true`. I guess that's why it generated a folder and put all the .cs files inside, since I want separate classes. I will try to customize it so I still get separate classes without the code generating a folder for them.

**Comment by mganss** [comment]
This is because you might have multiple classes with the same name but in different namespaces.
